platform/linux-generic/include/odp_packet_dpdk.h

```diff
@@ -49,16 +49,17 @@ typedef union {
 typedef struct {
 	odp_pool_t pool;		  /**< pool to alloc packets from */
 	struct rte_mempool *pkt_pool;	  /**< DPDK packet pool */
-	odp_pktio_capability_t	capa;	  /**< interface capabilities */
 	uint32_t data_room;		  /**< maximum packet length */
-	uint16_t mtu;			  /**< maximum transmission unit */
-	/** Use system call to get/set vdev promisc mode */
-	odp_bool_t vdev_sysc_promisc;
-	uint8_t port_id;		  /**< DPDK port identifier */
 	unsigned min_rx_burst;		  /**< minimum RX burst size */
 	odp_pktin_hash_proto_t hash;	  /**< Packet input hash protocol */
-	odp_bool_t lockless_rx;		  /**< no locking for rx */
-	odp_bool_t lockless_tx;		  /**< no locking for tx */
+	uint16_t mtu;			  /**< maximum transmission unit */
+	/** Use system call to get/set vdev promisc mode */
+	uint8_t vdev_sysc_promisc;
+	uint8_t lockless_rx;		  /**< no locking for rx */
+	uint8_t lockless_tx;		  /**< no locking for tx */
+	uint8_t port_id;			  /**< DPDK port identifier */
+	/* --- 34 bytes --- */
+	odp_pktio_capability_t	capa;	 /**< interface capabilities */
```

Comment:
pktio capability is big struct and it's same for all packets.  Is it really needs to be here? as well as uint16_t mtu? why not in pktio_entry_t ?

Reply:
A valid point, at least I don't see any reason why capability struct is stored inside each pktio implementation. MTU is used in the fast path so I wouldn't move that. Should I move the capability in this PR or create a new one?

Reply:
I think if it will be in this request it will make easy to review all changes at once.